src/dmd/dtemplate.d

```diff
@@ -1351,8 +1351,37 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
             {
                 Parameter fparam = fparameters[parami];
 
+                Type prmtype = fparam.type;
+                //printf("prmtype: %s\n", prmtype.toChars());
+                while(prmtype.ty == Tinstance)
+                {
+                    TypeInstance prmti = cast(TypeInstance) prmtype;
+                    //printf("prmti: %s\n", prmti.toChars());
+                    if(prmti.tempinst.findTempDecl(sc, null))
+                    {
+                        TemplateDeclaration td = prmti.tempinst.tempdecl.isTemplateDeclaration();
+                        if(td)
```

Comment:
You can put the declaration directly as condition `if( TemplateDeclaration td = prtmi....)`